barretenberg/cpp/src/barretenberg/ultra_honk/decider_verifier.cpp

```diff
@@ -45,32 +45,59 @@ template <typename Flavor> bool DeciderVerifier_<Flavor>::verify()
 
     auto sumcheck = SumcheckVerifier<Flavor>(
         static_cast<size_t>(accumulator->verification_key->log_circuit_size), transcript, accumulator->target_sum);
+    // Receive commitments to Libra masking polynomials
+    std::vector<Commitment> libra_commitments;
+    if constexpr (Flavor::HasZK) {
+        for (size_t idx = 0; idx < static_cast<size_t>(accumulator->verification_key->log_circuit_size); idx++) {
+            Commitment libra_commitment =
+                transcript->template receive_from_prover<Commitment>("Libra:commitment_" + std::to_string(idx));
+            libra_commitments.push_back(libra_commitment);
+        };
+    }
+    SumcheckOutput<Flavor> sumcheck_output;
 
-    auto [multivariate_challenge, claimed_evaluations, sumcheck_verified] =
+    sumcheck_output =
         sumcheck.verify(accumulator->relation_parameters, accumulator->alphas, accumulator->gate_challenges);
 
     // If Sumcheck did not verify, return false
-    if (sumcheck_verified.has_value() && !sumcheck_verified.value()) {
+    if (sumcheck_output.verified.has_value() && !sumcheck_output.verified.value()) {
         info("Sumcheck verification failed.");
         return false;
     }
 
-    const auto opening_claim = Shplemini::compute_batch_opening_claim(accumulator->verification_key->circuit_size,
-                                                                      commitments.get_unshifted(),
-                                                                      commitments.get_to_be_shifted(),
-                                                                      claimed_evaluations.get_unshifted(),
-                                                                      claimed_evaluations.get_shifted(),
-                                                                      multivariate_challenge,
-                                                                      Commitment::one(),
-                                                                      transcript);
+    BatchOpeningClaim<Curve> opening_claim;
+    if constexpr (!Flavor::HasZK) {
```

Comment:
looks ugly, but it's temporary. i think we could template Shplemini on Flavor and get rid of this mess